src/os/bluestore/BlueStore.cc

```diff
@@ -5142,6 +5142,7 @@ int BlueStore::queue_transactions(
   txc->onreadable = onreadable;
   txc->onreadable_sync = onreadable_sync;
   txc->oncommit = ondisk;
+  osr->qlock.lock();
```

Comment:
for simplicity i'd just do

 std::lock_guard<std::mutex> l<osd->qlock);

that wya it holds it through the end of hte function (including state_proc)